### PR TITLE
Add depth dimension as arg to compute_zmid

### PR DESCRIPTION
**Pull request by xylar**
This will make the function useful for computing `zMid` in other contexts (e.g. COMPASS), not just CMIP6 data prep.

### CODE REVIEW

**Comment by xylar** [comment]
@mark-petersen, with this change this function might be useful in COMPASS as a replacement for computing zMid in each test case from scratch.  I'm thinking specifically of: https://github.com/mark-petersen/MPAS-Model/blob/internal_tide_test_case/testing_and_setup/compass/ocean/internal_tide/3km/RK4/add_initial_state.py#L141-L147

File: conda_package/mpas_tools/ocean/depth.py [hunk 1]
```diff
@@ -14,16 +14,16 @@ def compute_depth(refBottomDepth):
 
     Parameters
     ----------
-    refBottomDepth : ``xarray.DataArray``
+    refBottomDepth : xarray.DataArray
```
Comment (mark-petersen) [inline conda_package/mpas_tools/ocean/depth.py:17]: What was the double backquote used for before, and why was it removed? I'm trying learn for myself.  Looks like a markdown thing the for documentation. I also just learned that a single backquote is an alias for `repl( )` in python, an unrelated fact.

**Review by mark-petersen — APPROVED** [review]
Looks good. If it works for you, please merge.

File: conda_package/mpas_tools/ocean/depth.py (hunk 1 above)
Comment (xylar) [reply conda_package/mpas_tools/ocean/depth.py:17]: There are in RestructuredText, which is a little different from markdown.

The double back-quotes are for literal strings, like single back-quotes in markdown.  But `sphinx` (the package that processes the documentation, including these docstrings) can link to `xarray` documentation if we don't use the back-quotes but not if we do.

I have never seen single back-quotes used in Python, nor have I ever used `repl()`.  Sounds fancy but potentially confusing...

**Pull request merged**